src/client/Microsoft.Identity.Client/Internal/Requests/ClientCredentialRequest.cs

```diff
@@ -56,7 +56,8 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
                         AuthenticationRequestParameters.AuthenticationScheme,
                         AuthenticationRequestParameters.RequestContext.CorrelationId,
                         TokenSource.Cache,
-                        AuthenticationRequestParameters.RequestContext.ApiEvent);
+                        AuthenticationRequestParameters.RequestContext.ApiEvent,
+                        CacheManager);
```

Comment:
>CacheManager [](http://example.com/codeflow?start=24&length=12)

i don't think we should return a CacheManager. plus the AT is already returned above in cachedAccessTokenItem, line 46. is there another way to access the tenant profiles?

Reply:
Another way would be to have a delegate in account instead as mentioned by @jmprieur in one of the comments above. But I have yet to evaluate how that can be done. We need the CacheManager to get the tenant profiles for IdTokens in the cache. I can't think of any other way.

Reply:
We can have the Request objects fetch the data needed for creating the tenant profiles, let me have a look...